### PR TITLE
Fix codegen

### PR DESCRIPTION
**Pull request by pawelprazak**
- Fix missing JsonFormat for PulumiAny
- Fix multi-line @deprecated argument in codegen
- Mangle property names in codegen with reserved package names
- Fix codegen type unescaping method
- Fix codegen type mapper for malformed named types with not leading slash

new Packages scripts functionality

### CODE REVIEW

**Review by lbialy — CHANGES_REQUESTED** [review]

File: core/src/main/scala/besom/types.scala [hunk 1]
```diff
@@ -11,9 +11,8 @@ import scala.language.implicitConversions
 import scala.util.Try
 
 object types:
-  // TODO: replace these stubs with proper implementations
   private object Opaques:
-    opaque type PulumiAny = besom.json.JsValue
+    type PulumiAny = besom.json.JsValue
     object PulumiAny:
       given Encoder[PulumiAny] = Encoder.jsonEncoder
       given Decoder[PulumiAny] = Decoder.jsonDecoder
```
[the remark this answers is not in the document]
Comment (pawelprazak) [reply core/src/main/scala/besom/types.scala:18]: Pulumi has two types `pulumi.json#/Any` and `pulumi.json#/Json`

File: core/src/main/scala/besom/types.scala [hunk 2]
```diff
@@ -11,9 +11,8 @@ import scala.language.implicitConversions
 import scala.util.Try
 
 object types:
-  // TODO: replace these stubs with proper implementations
   private object Opaques:
```
[the remark this answers is not in the document]
Comment (pawelprazak) [reply core/src/main/scala/besom/types.scala:14]: done

File: core/src/main/scala/besom/internal/Config.scala [hunk 3]
```diff
@@ -89,7 +89,7 @@ class Config private (
     *   an optional configuration or secret value of the requested type, either from the configuration bag or from the environment, or
     *   default value
     */
-  def getOrDefault[A: ConfigValueReader](
+  private [internal] def getOrDefault[A: ConfigValueReader](
```
[the remark this answers is not in the document]
Comment (pawelprazak) [reply core/src/main/scala/besom/internal/Config.scala:92]: fixed the formatting

File: core/src/main/scala/besom/types.scala (hunk 1 above)
[the remark this answers is not in the document]
Comment (pawelprazak) [reply core/src/main/scala/besom/types.scala:18]: changed by dropping opaque

**Review by lbialy — APPROVED** [review]

**Pull request merged**